File: trading/order.js
/**
 * 订单管理 - Node.js 版本
 */

const OrderType = Object.freeze({
  MARKET: 'MARKET',
  LIMIT: 'LIMIT',
  STOP_MARKET: 'STOP_MARKET',
  STOP_LIMIT: 'STOP_LIMIT'
});

const OrderSide = Object.freeze({
  BUY: 'BUY',
  SELL: 'SELL'
});

const OrderStatus = Object.freeze({
  NEW: 'NEW',
  PARTIALLY_FILLED: 'PARTIALLY_FILLED',
  FILLED: 'FILLED',
  CANCELED: 'CANCELED',
  REJECTED: 'REJECTED',
  PENDING_CANCEL: 'PENDING_CANCEL'
});

class Order {
  /**
   * @param {Object} config
   * @param {string} config.orderId
   * @param {string} config.symbol
   * @param {OrderSide} config.side
   * @param {OrderType} config.type
   * @param {number} config.quantity
   * @param {number} config.price
   * @param {number} config.stopPrice
   * @param {OrderStatus} config.status
   * @param {Date} config.createTime
   */
  constructor(config = {}) {
    this.orderId = config.orderId;
    this.symbol = config.symbol;
    this.side = config.side;
    this.type = config.type;
    this.quantity = config.quantity;
    this.price = config.price;
    this.stopPrice = config.stopPrice;
    this.status = config.status || OrderStatus.NEW;
    this.createTime = config.createTime || new Date();

    this.filledQuantity = 0;
    this.avgPrice = null;
    this.updateTime = null;
  }

  get isFilled() {
    return this.status === OrderStatus.FILLED;
  }

  get isActive() {
    return [OrderStatus.NEW, OrderStatus.PARTIALLY_FILLED].includes(this.status);
  }
}

module.exports = {
  Order,
  OrderType,
  OrderSide,
  OrderStatus
};

File: BinanceChanQuant/src/main/java/hft/executor/Order.java
package hft.executor;

import java.util.concurrent.atomic.AtomicLong;

/**
 * Order - Order entity for HFT System
 */
public class Order {
    private static final AtomicLong idGenerator = new AtomicLong(0);

    public final String id;
    public final String symbol;
    public final Order.Side side;
    public final Order.Type type;
    public final double price;  // 0 for market orders
    public final double size;
    public volatile double filled = 0;
    public volatile double avgPrice = 0;
    public volatile Order.Status status = Order.Status.PENDING;
    public final long createdAt;
    public volatile long updatedAt;
    public volatile long binanceOrderId = 0;

    public Order(String symbol, Order.Side side, Order.Type type, double price, double size) {
        this.id = "ord_" + idGenerator.incrementAndGet();
        this.symbol = symbol;
        this.side = side;
        this.type = type;
        this.price = price;
        this.size = size;
        this.createdAt = System.currentTimeMillis();
        this.updatedAt = this.createdAt;
    }

    public boolean isFilled() {
        return status == Order.Status.FILLED;
    }

    public boolean isOpen() {
        return status == Order.Status.OPEN || status == Order.Status.PARTIALLY_FILLED;
    }

    public void updateStatus(Order.Status newStatus) {
        this.status = newStatus;
        this.updatedAt = System.currentTimeMillis();
    }

    public enum Side { BUY, SELL }
    public enum Type { MARKET, LIMIT, STOP_MARKET }
    public enum Status { PENDING, OPEN, PARTIALLY_FILLED, FILLED, CANCELLED, REJECTED }
}
